homeassistant/helpers/event.py

```diff
@@ -881,6 +882,169 @@ def async_track_state_change_filtered(
     return tracker
 
 
+class _TrackStateReportedFiltered:
```

Comment:
I'd suggest to rename the existing class to `_TrackStateEventFiltered` and make it take the event type as a parameter.